promql/engine.go

```diff
@@ -997,29 +1005,32 @@ func (ev *evaluator) rangeEval(f func([]parser.Value, *EvalNodeHelper) Vector, e
 		mat = append(mat, ss)
 	}
 	ev.currentSamples = originalNumSamples + mat.TotalSamples()
-	return mat
+	return mat, warnings
 }
 
 // evalSubquery evaluates given SubqueryExpr and returns an equivalent
 // evaluated MatrixSelector in its place. Note that the Name and LabelMatchers are not set.
-func (ev *evaluator) evalSubquery(subq *parser.SubqueryExpr) *parser.MatrixSelector {
-	val := ev.eval(subq).(Matrix)
-	vs := &parser.VectorSelector{
-		Offset: subq.Offset,
-		Series: make([]storage.Series, 0, len(val)),
-	}
-	ms := &parser.MatrixSelector{
-		Range:          subq.Range,
-		VectorSelector: vs,
-	}
-	for _, s := range val {
+func (ev *evaluator) evalSubquery(subq *parser.SubqueryExpr) (*parser.MatrixSelector, storage.Warnings) {
```

Comment:
In a big PR like this, it's best not to make incidental style changes like in this function. It makes the diff bigger than it needs to be.

Reply:
Yea agree, especially that  the logic is different: Series: make([]storage.Series, 0, len(mat)),